app/src/main/java/com/neo/speaktouch/utils/VibrationUtil.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Vibration utility wrapper.
+ *
+ * Copyright (C) 2023 Irineu A. Silva.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.neo.speaktouch.utils
+
+import android.content.Context
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+
+class VibrationUtil(private val vibrator: Vibrator) {
+
+    @Suppress("deprecation")
+    constructor(context: Context) : this(
+        context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
+    )
+
+    fun vibrateEffectHeavyClick() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            vibrator.vibrate(
+                VibrationEffect.createPredefined(
+                    VibrationEffect.EFFECT_HEAVY_CLICK
+                )
+            )
+        } else {
```

Comment:
We can't forgot about API 33 change; otherwise, vibration on Android 13+ won't work at all.

<code>
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            vibrator.vibrate(
                VibrationEffect.createPredefined(
                    VibrationEffect.EFFECT_HEAVY_CLICK
                ),
                VibrationAttributes.createForUsage(
                    VibrationAttributes.USAGE_ACCESSIBILITY
                )
            )
        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            vibrator.vibrate(
                VibrationEffect.createPredefined(
                    VibrationEffect.EFFECT_TICK
                )
            )
        }  else {
        </code>

Reply:
Okay, I didn't know about this change.